app/controllers/sks_users_controller.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpContext } from '@adonisjs/core/http'
+import SksUser from '#models/sks_user'
+import { DateTime } from 'luxon'
+
+//this value determines the period over which the trend will be counted (multiply the value by 5 minutes)
+const trendDelta = 3
+
+enum Trend {
+  INCREASING = 'INCREASING',
+  DECREASING = 'DECREASING',
+  STABLE = 'STABLE',
+}
+
+export default class SksUsersController {
+  /**
+   * Display the latest SKS users record based on time
+   */
+  async latest({ response }: HttpContext) {
+    try {
+      const currentTime = DateTime.now().setZone('Europe/Warsaw').toSQL()
+      if (currentTime === null) {
+        return response.status(500).json({ message: 'Failed to convert time to sql format' })
+      }
+
+      const latestData = await SksUser.query()
+        .where('externalTimestamp', '<', currentTime)
+        .orderBy('externalTimestamp', 'desc')
+        .offset(1) //get the second-latest record to ensure the data is already scraped
+        .first()
+      if (latestData === null) {
+        return response
+          .status(404)
+          .json({ message: 'Could not find the matching data in database' })
+      }
+
+      let trend: Trend
```

Comment:
So in a normal script I would extract the below logic to separate `getTrend` func or something - but I don't know what are the conventions in adonis and its controllers. Maybe it's better to leave it as it is to have a nice looking single controller

Reply:
U got me there, actually I thought about using a function, but was to lazy to do it 😭